src/main/java/io/gravitee/policy/callout/CalloutHttpPolicy.java

```diff
@@ -245,9 +268,15 @@ public void handle(Buffer body) {
                         }
                     }).exceptionHandler(throwable -> {
 
-                        if (configuration.isExitOnError()) {
-                            // exit chain only if policy ask ExitOnError
-                            onError.accept(PolicyResult.failure(CALLOUT_HTTP_ERROR, throwable.getMessage()));
+                        // exit and validate ExitOnError only if ignore exceptions are false
+                        if (!configuration.isContinueOnTimeout()) {
```

Comment:
`exceptionHandler` is not specific to timeouts.

I'm still trying to understand what is the difference between the `isContinueOnTimeout` and `isExitOnError`.

Maybe it could be difficult to understand when to use one or the other.
Why not adding an information within the context to tell that we are getting a timeout, and then, let the user to use the `errorCondition` to specify if he wants to continue or not.

Let's imagine that the gateway is getting a timeout, in that case, we are adding a new information in the execution context to say that we get a timeout.
And in the error condition, the user can fill it with something like `{! #calloutResponse.timeout and #calloutResponse.status >= 400 and #calloutResponse.status <= 599}`.

That wait, we are not adding multiple checkbox for each cases and we still benefit from the errorCondition to specify the behavior.
WDYT?

Reply:
David,

Yes, I agree that it can cause confusion, but think about it, I need the flow to stop, if the request for the callout receives errors: 401, 403 or 422, and only in this case, then isExitOnError works well, but what if the request suffer a timeout or even a connection problem, the flow in my case cannot stop, it must continue, and therefore it has this control, ignore these cases, isExitOnError stops these errors. Your suggestion, to treat this condition can work well and put more power in decision making. But did I manage to explain why I need this situation?

Reply:
Hi @klinux,

Yes, I got your use-case, it's clear to me.
I'm still thinking that centralizing the exit-on-error-condition is the best way to avoid to multiply different conditions to drive the behavior.

Reply:
In my case, I need that flow ignore read timeouts, connection timeouts and only stop flow with an specific status code, I put this checkbox to threat that, but I agree, put it into errorCondition is much better than checkbox, is more powerfull.

Reply:
Yes, I got you and it makes sense to me.

So we both agree that defining condition for timeout is a better option :)

Reply:
Yes, definitely :)

Reply:
Can I close this PR? I think you think a completely diferent implementation about it, I'm wright?

Reply:
I think so, yes. 

Your use-case is interesting but we must keep configuration the most easy and understandable as possible.